pkg/ingester/ingester.go

```diff
@@ -312,7 +312,9 @@ type Ingester struct {
 	// Metrics shared across all per-tenant shippers.
 	shipperMetrics *shipperMetrics
 
-	subservices  *services.Manager
+	subservicesBeforePartitionReader       *services.Manager
+	subservicesAfterIngesterRingLifecycler *services.Manager
```

Comment:
WDYT about `subservicesBeforePartitionReader` => `subservicesForPartitionReplay` to indicate that these are running when we're doing partition replay?

(Then we could also try `subservicesAfterIngesterRingLifecycler` => `subservicesAfterPartitionReplayAndIngesterRingLifecycler`, but that seems bit long).

Reply:
> WDYT about subservicesBeforePartitionReader => subservicesForPartitionReplay to indicate that these are running when we're doing partition replay?

I have no preference, so I named it as you suggested: https://github.com/grafana/mimir/pull/7769/commits/35288f1ec0b6fd3cb8fe216ed099e30c7e6bf184

> (Then we could also try subservicesAfterIngesterRingLifecycler => subservicesAfterPartitionReplayAndIngesterRingLifecycler, but that seems bit long).

Skipped this. Look too long.